utils/streams.js

```diff
@@ -0,0 +1,205 @@
+    const readableStream = fs.createReadStream(filePath, 'utf8');
+    
+    const fileName = createFilePath(filePath);
+    const writableStream = fs.createWriteStream(fileName);
+    readableStream
+      .pipe(through(csvToJsonStreamTransform, () => {
+        console.log(`${fileName} was generated`);
+      }))
+      .pipe(writableStream);
+  }
+  else console.log('Please specify correct file path');
+}
+
+/* generate test files */
+function writeDataToStream(data, writerableStream, count) {
+  if (count === 0) return;
+  writerableStream.write(data, function() {
+    writeDataToStream(data, writerableStream, --count);
+  });
+}
+
+function getFolderPath(path) {
+  return path[path.length - 1] === "/" ? path : `${path}/`;
+}
+
+function createFileName(filePath, extension, prefixName) {
+  const pathJsonFile = path.parse(filePath);
+  if (extension) {
+    pathJsonFile.ext = extension;
+  }
+  pathJsonFile.base = `${pathJsonFile.name}${prefixName}${pathJsonFile.ext}`;
+  return path.format(pathJsonFile);
+}
+
+function generateTestCSS(path = './data/css') {
+  fs.readdir(path, (err, files) => {
+    files.forEach(file => {
+      console.log(createFileName(`${getFolderPath(path)}${file}`, null, '_large'));
+      const writerableStream = fs.createWriteStream(createFileName(`${getFolderPath(path)}${file}`, null, '_large'));
+      let readableStream = fs.createReadStream(`${getFolderPath(path)}${file}`, 'utf8');
+      let data = '';
+      readableStream.on('data', function (chunk) {
+        data += chunk; //only for small files
+      });
+      readableStream.on('end', function () {
+        writeDataToStream(data, writerableStream, 400000);
+      });
+    });
+  });
+}
+
+/* CSS Bundler */
+function asyncOrderedWrite(cssFiles, writerableStream, pathFolder, additionalStream = () => {}) {
+  if (cssFiles.length === 0) {
+    additionalStream(writerableStream);
+    return;
+  }
+  let remainingCssFiles = cssFiles.splice(1);
+  let cssFile = cssFiles[0];
+  const readableStream = fs.createReadStream(`${pathFolder}${cssFile}`, 'utf8');
+  
+  readableStream.on('data', function(chunk) {
+    writerableStream.write(chunk + "\n");
+  });
+  
+  readableStream.on('end', function() {
+    asyncOrderedWrite(remainingCssFiles, writerableStream, pathFolder, additionalStream);
+  });
+}
+
+async function combineCSSFiles(pathFolder, resultFileName, writerableStream, additionalStream) {
+  fs.readdir(pathFolder, (err, files) => {
+    const cssFiles = files.filter(file => path.extname(file) === '.css' && file !== resultFileName);
+    asyncOrderedWrite(cssFiles, writerableStream, pathFolder, additionalStream);
+  });
+}
+
+function getCssFromWeb(writerableStream) {
+  request(BASE_CSS_URL).on('data', function(chunk) {
+    writerableStream.write(chunk);
+  });
+}
+
+function cssBundler(path, resultFileName = 'bundle.css') { /* 8, babel-node -- ./utils/streams.js -a cssBundler -p ./data/css */
```

Comment:
Looks much better now but it's a bit complicated. Creating a queue of streams could be helpful not only for this task and it would be better to extract this functionality to separate module (or use existing one but writing a new one is a more challenging and interesting task). It will also help a lot with readability.

And in result it will look similar to this:
```js
const writableStream = fs.createWriteStream(`${getFolderPath(folderPath)}${resultFileName}`);
multiStream([
  ...getCssFilesFromPath(folderPath).map(fileName => fs.createReadStream(fileName, 'utf8')),
  request(BASE_CSS_URL),
]).pipe(writableStream);
```

Reply:
cool, thx